### PR TITLE
s3 abi

### PR DESCRIPTION
**Pull request by NemeZZiZZ**

### CODE REVIEW

**Review by evahteev — APPROVED** [review]

File: src/components/VerifyABI/VerifyABI.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+export const VerifyABI: FC<VerifyABIProps> = ({ className }) => {
+  const [currentFiles, setCurrentFiles] = useState<FileList | null>(null)
+  const [currentError, setCurrentError] = useState<string | null>(null)
+  const [isReady, setIsReady] = useState(false)
+  const [isSent, setIsSent] = useState(false)
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+
+    if (!process.env.NEXT_PUBLIC_ABI_VERIFY_URL) {
```
Comment (Kharabet) [inline src/components/VerifyABI/VerifyABI.tsx]: process.env.NEXT_PUBLIC_ABI_VERIFY_URL

there no sense to calc this env var inside the component. Would like it to be a const outside the component cause it is known on the container start and won't change

File: src/pages/api/verifyABI.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  },
+}
+
+console.log(
+  process.env.NEXT_PUBLIC_S3_REGION,
+  process.env.NEXT_PUBLIC_S3_ACCESS_KEY,
+  process.env.NEXT_PUBLIC_S3_SECRET_KEY,
+  process.env.NEXT_PUBLIC_S3_BUCKET,
+  process.env.NEXT_PUBLIC_S3_PATH
+)
```
Comment (Kharabet) [inline src/pages/api/verifyABI.js]: would be good to do smth like https://github.com/dex-guru/dexguru-frontend-bloomberg/blob/guruv2-prod/src/services/flow/client.ts#L13-L16

the whole route handle will not work without these env vars anyway so you can throw Errors

File: src/pages/api/verifyABI.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+const s3 = new S3Client({
+  region: process.env.NEXT_PUBLIC_S3_REGION ?? '',
+  credentials: {
+    accessKeyId: process.env.NEXT_PUBLIC_S3_ACCESS_KEY ?? '',
+    secretAccessKey: process.env.NEXT_PUBLIC_S3_SECRET_KEY ?? '',
+  },
+})
```
Comment (Kharabet) [inline src/pages/api/verifyABI.js]: and then you will get rid of these fallbacks to `?? ''`

**Review by Kharabet — APPROVED** [review]
Left minor comments. could be fixed later in favor of HOT hackathon delivery

**Review by Kharabet — APPROVED** [review]

**Pull request merged**